src/hooks/useNetInfo.ts

```diff
@@ -0,0 +1,14 @@
+import NetInfo from '@react-native-community/netinfo';
+import {useEffect, useState} from 'react';
+
+export function useNetInfo() {
+  const [networkStatus, setNetworkStatus] = useState<boolean>(false);
+
+  useEffect(() => {
+    NetInfo.addEventListener((state) => {
+      setNetworkStatus(state.isConnected ? true : false);
+    });
```

Comment:
```suggestion
    const unsubscribe = NetInfo.addEventListener((state) => {
      setNetworkStatus(state.isConnected ? true : false);
    });
    
    return () => {
        unsubscribe();
    }
```

For reference: https://reactjs.org/docs/hooks-effect.html#effects-with-cleanup